design/reboot-interface.md

```diff
@@ -89,78 +88,76 @@ section of the BareMetalHost status. This records a time after which the Host
 was last booted using the current image. Processes running prior to this time
 may be assumed to have been stopped.
 
-To provide a declarative interface for performing a reboot, a new
-BareMetalHostRebootRequest CRD will be created. The spec of this custom
-resource will contain only a reference to a BareMetalHost. The status will
-contain timestamps for when the Host was powered off and on again. Once the
-power off time is present in the status, the host has been stopped. Each
-RebootRequest will be owned by the Host it refers to, so that if the Host is
-deleted all associated RebootRequest CRs will be cleaned up.
+A new date-time field, ``pendingRebootSince``, will be added to the ``provisioning``
+section of the BareMetalHost status. This records a time before which the Host
+was last requested to reboot (because we cannot trust any value from the user,
+who even if well intentioned, may have created the timestamp on a machine that 
+was not synchronised with the cluster or has a different timezone).
+
+Since the user interface requirements are still unclear, we will follow 
+standard practices of using an annotation to trigger reboots.  
+
+The basic annotation form ( ``reboot.metal3.io`` ) triggers the controller to
+power cycle the Host.  This form has set-and-forget semantics and the 
+controller removes the annotation once it restores power to the Host. 
+
+An advanced form ( ``reboot.metal3.io/{key}`` ) instructs the controller hold 
+the Host in a ``PoweredOff`` state so that the caller can perform any required
+actions while the node is in a known safe state.   Callers indicate to the 
+controller that they are ready to continue by removing the annotation with their 
+unique ``{key}}`` suffix.
```

Comment:
I'm not sure how I feel about this part. It makes it seem a lot more imperative. I know we're trying to avoid race conditions, but I wonder if that means the controller that needs to control the process should just use the existing online flag and manage the reboot itself.

Reply:
It's not ideal but still better than nothing IMHO. This way allows multiple things to do that independently without interfering with each other or with something trying to set the power state permanently.

Reply:
I'm struggling to imagine why we would have multiple things asking to reboot a host at the same time. Wouldn't it be better for those other things to add the remediation annotation to the node or machine and let the new controller handle it?